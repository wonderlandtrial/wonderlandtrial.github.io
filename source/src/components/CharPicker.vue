<template>
<v-dialog
  v-model="model"
  max-width="500px"
  origin="top center">
  <v-card>
    <v-card-title>Character Picker</v-card-title>
    <v-card-text>

      <v-select
        label="Select a character..."
        :items="characters"
        v-model="selectedChar"
        item-text="name"
      >
        <template slot="selection" slot-scope="data">
          <char-avatar :size="24" v-model="data.item" />
          {{data.item.name}}
        </template>

        <template slot="item" slot-scope="data">
          <char-avatar :size="48" v-model="data.item" />
          {{data.item.name}}
        </template>
      </v-select>

      <v-text-field
        label="Level"
        v-model="level"
      ></v-text-field>

      <v-text-field
        label="Note"
        v-model="note"
      ></v-text-field>

    </v-card-text>
    <v-card-actions>
      <v-btn color="primary" flat @click="selectChar">OK</v-btn>
      <v-btn flat @click="model=false">CANCEL</v-btn>
    </v-card-actions>
  </v-card>
</v-dialog>
</template>

<script>
import TeamMember from '@/models/TeamMember'
import CharAvatar from '@/components/CharAvatar'

export default {
  name: 'CharPicker',
  components: { CharAvatar },
  props: ['value', 'characters'],
  data() {
    return {
      model: false,
      selectedChar: null,
      level: 60,
      note: ''
    }
  },
  watch: {
    value(val) {
      this.model = val
    },
    model(val) {
      this.$emit('input', val)
    }
  },
  mounted() {
    this.model = this.value
  },
  methods: {
    selectChar() {
      this.model = false

      const selectedMember = new TeamMember(this.selectedChar, this.level, this.note)
      this.$emit('selected', {...selectedMember})
    }
  }
}
</script>
