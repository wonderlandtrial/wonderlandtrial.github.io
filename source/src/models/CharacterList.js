import Character from './Character'

export default {
  gold: [
    new Character('Crushfang', 'gold'),
    new Character('Dylan', 'gold')
  ],
  black: [
    new Character('Fatima', 'black'),
    new Character('Ned', 'black')
  ],
  white: [
    new Character('Angelia', 'white'),
    new Character('Aosta', 'white')
  ],
  advisor: [
    new Character('Crushfang', 'gold'),
    new Character('Dylan', 'gold'),
    new Character('Fatima', 'black'),
    new Character('Ned', 'black'),
    new Character('Angelia', 'white'),
    new Character('Aosta', 'white')
  ],
  monster: [
    new Character('monster 1', 'monster'),
    new Character('monster 2', 'monster')
  ]
}
