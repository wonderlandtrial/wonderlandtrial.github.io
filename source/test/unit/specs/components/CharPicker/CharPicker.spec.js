import { test } from '@/test'

import CharPicker from '@/components/CharPicker2'
import CharacterList from '@/models/CharacterList'

test('CharPicker', ({ mount, localVue }) => {
  it('rendered empty picker', ()=> {
    const wrapper = mount(CharPicker, { localVue })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('render white picker', ()=> {
    const wrapper = mount(CharPicker, {
      propsData: {
        characters: CharacterList.white
      },
      localVue
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
