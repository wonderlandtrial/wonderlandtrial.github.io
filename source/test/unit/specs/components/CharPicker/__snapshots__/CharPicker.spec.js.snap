// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharPicker render white picker 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog"
         style="max-width: 500px; display: none;"
    >
      <div class="card"
           style="height: auto;"
      >
        <div class="card__title">
          Character Picker
        </div>
        <div class="card__text">
          <div tabindex="0"
               data-uid="22"
               role="combobox"
               class="input-group input-group--append-icon input-group--text-field input-group--select primary--text"
          >
            <label>
              Select a character...
            </label>
            <div class="input-group__input">
              <div class="input-group__selections"
                   style="overflow: hidden;"
              >
                <input disabled="disabled"
                       tabindex="-1"
                       class="input-group--select__autocomplete"
                       style="display: none;"
                >
              </div>
              <div class="menu"
                   style="display: inline-block;"
              >
                <div class="menu__content menu__content--select"
                     style="max-height: 300px; min-width: 0px; max-width: auto; top: 18px; left: 0px; z-index: 0; display: none;"
                >
                  <div class="card"
                       style="height: auto;"
                  >
                    <div class="list">
                      <div>
                        <a class="list__tile list__tile--link">
                          <div class="avatar"
                               style="height: 48px; width: 48px;"
                          >
                            <img src="/static/assets/potraits/angelia_potrait.png"
                                 alt="Angelia"
                            >
                          </div>
                          Angelia
                        </a>
                      </div>
                      <div>
                        <a class="list__tile list__tile--link">
                          <div class="avatar"
                               style="height: 48px; width: 48px;"
                          >
                            <img src="/static/assets/potraits/aosta_potrait.png"
                                 alt="Aosta"
                            >
                          </div>
                          Aosta
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <i aria-hidden="true"
                 class="icon material-icons input-group__append-icon input-group__icon-cb"
              >
                arrow_drop_down
              </i>
            </div>
            <div class="input-group__details">
            </div>
          </div>
          <div class="input-group input-group--dirty input-group--text-field primary--text">
            <label>
              Level
            </label>
            <div class="input-group__input">
              <input tabindex="0"
                     aria-label="Level"
                     type="text"
              >
            </div>
            <div class="input-group__details">
            </div>
          </div>
        </div>
        <div class="card__actions">
          <button type="button"
                  class="btn btn--flat primary--text"
          >
            <div class="btn__content">
              OK
            </div>
          </button>
          <button type="button"
                  class="btn btn--flat"
          >
            <div class="btn__content">
              CANCEL
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>

`;

exports[`CharPicker rendered empty picker 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog"
         style="max-width: 500px; display: none;"
    >
      <div class="card"
           style="height: auto;"
      >
        <div class="card__title">
          Character Picker
        </div>
        <div class="card__text">
          <div tabindex="0"
               data-uid="6"
               role="combobox"
               class="input-group input-group--append-icon input-group--text-field input-group--select primary--text"
          >
            <label>
              Select a character...
            </label>
            <div class="input-group__input">
              <div class="input-group__selections"
                   style="overflow: hidden;"
              >
                <input disabled="disabled"
                       tabindex="-1"
                       class="input-group--select__autocomplete"
                       style="display: none;"
                >
              </div>
              <div class="menu"
                   style="display: inline-block;"
              >
                <div class="menu__content menu__content--select"
                     style="max-height: 300px; min-width: 0px; max-width: auto; top: 18px; left: 0px; z-index: 0; display: none;"
                >
                  <div class="card"
                       style="height: auto;"
                  >
                    <div class="list">
                      <div disabled="disabled"
                           class="text--disabled"
                      >
                        <a disabled="disabled"
                           class="list__tile list__tile--link list__tile--disabled"
                        >
                          <div class="avatar"
                               style="height: 48px; width: 48px;"
                          >
                            <img>
                          </div>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <i aria-hidden="true"
                 class="icon material-icons input-group__append-icon input-group__icon-cb"
              >
                arrow_drop_down
              </i>
            </div>
            <div class="input-group__details">
            </div>
          </div>
          <div class="input-group input-group--dirty input-group--text-field primary--text">
            <label>
              Level
            </label>
            <div class="input-group__input">
              <input tabindex="0"
                     aria-label="Level"
                     type="text"
              >
            </div>
            <div class="input-group__details">
            </div>
          </div>
        </div>
        <div class="card__actions">
          <button type="button"
                  class="btn btn--flat primary--text"
          >
            <div class="btn__content">
              OK
            </div>
          </button>
          <button type="button"
                  class="btn btn--flat"
          >
            <div class="btn__content">
              CANCEL
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>

`;
